File: Programmers/LV1/정리 필요/x만큼간격이 있는 n개의 숫자.py
# def solution(x,n):
#     answer = []
#
#     if x < 0:
#         for i in range(x,x*n-1,x):
#             answer.append(i)
#     elif x == 0:
#         for i in range(0,n+1,0):
#             answer.append(i)
#     else:
#         for i in range(x,x*n+1,x):
#             answer.append(i)
#     return answer
#
#
# print(solution(-4,2))
# print(solution(0,2))
# print(solution(4,3))
# print(solution(2,5))

def solution(x,n):
    return [x*i for i in range(1,n+1) ]

print(solution(3,5))

File: Programmers/LV1/JAVA/x만큼간격이있는n개의숫자.java
package Programmers.level1;

public class x만큼간격이있는n개의숫자 {

	public static void main(String[] args) {

		System.out.println(solution(2, 5));

	}

	public static long[] solution(int x, int n) {

		long[] answer = new long[n];

		answer[0] = x;

		for (int i = 1; i < n; i++) {
			answer[i] = answer[i - 1] + x;
		}

		return answer;
	}
	
	//다른풀이
	public static long[] solution2(int x, int n) {
		long[] answer = new long[n];
		for (int i = 0; i < n; i++) {
			answer[i] = x * (i + 1);
		}
		return answer;
	}

}
